### PR TITLE
[IMP] stock: prepare for Barcode app new design

### PR DESCRIPTION
**Pull request by svs-odoo**
Remove useless `state` field in the `stock_picking_kanban` view to make xpath targeting this field easier in the `stock_barcode` module.

task-3929927

Enterprise PR: odoo/enterprise#63727
Upgrade PR: odoo/upgrade#6249

### CODE REVIEW

File: addons/web/static/src/webclient/barcode/barcode_scanner.js [hunk 1]
```diff
@@ -10,13 +10,12 @@ import { CropOverlay } from "./crop_overlay";
 import { Component, onMounted, onWillStart, onWillUnmount, useRef, useState } from "@odoo/owl";
 import { _t } from "@web/core/l10n/translation";
 
-export class BarcodeDialog extends Component {
-    static template = "web.BarcodeDialog";
+export class BarcodeVideoScanner extends Component {
```
Comment (rfr-odoo) [inline addons/web/static/src/webclient/barcode/barcode_scanner.js:13]: Split the two components into individual files

File: addons/web/static/src/webclient/barcode/barcode_scanner.xml [hunk 2]
```diff
@@ -2,9 +2,21 @@
...
+            <div t-else="" t-ref="videoPreview"
+                 class="h-100 d-flex flex-column justify-content-center align-items-center gap-1">
+                <i class="fa fa-2x fa-camera text-muted"></i>
+                <strong>Unable to access camera</strong>
+                <span class="text-muted">Check your browser permissions</span>
+            </div>
         </Dialog>
     </t>
+
+    <t t-name="web.BarcodeVideoScanner">
```
Comment (rfr-odoo) [inline addons/web/static/src/webclient/barcode/barcode_scanner.xml:17]: Split the two templates into individual XML files (associated to JS file name)

File: addons/web/static/src/webclient/barcode/barcode_video_scanner.xml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates xml:space="preserve">
+    <t t-name="web.BarcodeVideoScanner">
+        <CropOverlay onResize.bind="this.onResize" isReady="state.isReady">
+            <video t-ref="videoPreview" muted="true" autoplay="true" playsinline="true" class="w-100 h-100"/>
+        </CropOverlay>
+    </t>
+</templates>
```
Comment (rfr-odoo) [inline addons/web/static/src/webclient/barcode/barcode_video_scanner.xml:8]: Missing return line

File: addons/web/static/src/webclient/barcode/barcode_dialog.xml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+                                 onError.bind="onError" onResult.bind="onResult"/>
+            <div t-else="" t-ref="videoPreview"
+                 class="h-100 d-flex flex-column justify-content-center align-items-center gap-1">
+                <i class="fa fa-2x fa-camera text-muted"></i>
+                <strong>Unable to access camera</strong>
+                <span class="text-muted">Check your browser permissions</span>
+            </div>
+        </Dialog>
+    </t>
+</templates>
```
Comment (rfr-odoo) [inline addons/web/static/src/webclient/barcode/barcode_dialog.xml:16]: Missing return line

**Comment by svs-odoo** [comment]
@odoo/rd-framework-py I need somebody to review [this commit](https://github.com/odoo/odoo/pull/167942/commits/311ce80a88ae22261ccd9e98a5b2c4a982f71ce1)

@odoo/rd-security Security check is red because of Markup use in the JS, the markuped value is the help got from a view (rendered as plain text if not markuped)

**Comment by svs-odoo** [comment]
@xmo-odoo Indeed, it's the wrong PR, I was thinking of the [Enterprise PR](https://github.com/odoo/enterprise/pull/63727) :confused:

**Comment by amoyaux** [comment]
robodoo r+

**Comment by amoyaux** [comment]
robodoo rebase-ff

**Pull request closed without merging**